jsonevtloop_jsonschema_V2.txt

```diff
@@ -0,0 +1,153 @@
+
+	-----------------------------------------
+	Description of the jsonevtloop and schema
+	-----------------------------------------
+
+Purpose: JSON-based asynchronous API for signal-cli.
+
+This mode is not intended to be used directly from the command line, but to 
+provide an easy use API for other scripting languages.
+
+signal-cli listens for JSON requests on stdin. Each JSON request must be
+a single valid JSON object (enclosed in {}), followed by LF. The JSON text
+itself hence must not contain linefeeds (only escaped as \n and no pretty
+printing enabled).
+
+When incoming messages are received from the signal back-end, they are
+in turned JSON encoded and sent out on stdout.
+
+Any error messages or debug information is sent out on stderr as expected.
+
+
+
+JSON Schema Version 2 
+
+
+	JSON Requests are read from STDIN in "jsonevtloop" mode and the Reponses are
+	sent to STDOUT in "jsonevtloop" mode
+	
+	Generic Request attributes valid for ALL requests
+	{
+		"reqType": "<request-type>",				// MANDATORY: Name of request type
+		"reqID": <integer or string>				// Optional: reqID that will be present in the 
+													// response message for async callers to use as transaction id
+													// (note: for brevity omitted in the examples below)
+	}
+	Generic Response attributes valid for ALL responses
+	{
+		"respType": "<response-type>",				// Name of response type
+		"apiVer": 2,								// API Version 2
+		"reqID": <object>,							// Request Identifier as present in the request object
+		"status": "<ok|error>",						// Whether the request was a success of failure
+		"message": "<error-message>"				// In case of status="error", this contains any error message
+	}
+
+	
+	Send Message Request (direct message to another person)
+	{
+		"reqType": "send_message",
+		"recipientNumber": "<number>",
+		"messageBody": "<message-body-text>",
+		"attachments": [ "<file-name1>", ... ]
+	}
```

Comment:
For sending messages I'd prefer a format that is similar to the format of received messages and also more flexible.
Something like this:
```json
        {
	  "apiVer": 2,
	  "type": "send_message",
	  "recipient": {"number": "<number>"},
	  "message": "<message-body-text>",
	  "dataMessage": {
	    "message": "😋😋😋",
	    "attachments": [
	      {
	        "contentType": "image/jpeg",
	        "filename": "signal-2020-02-09-122846.jpeg"
	      },
	      {
	        "filename": "signal-2020-02-09-122846.jpeg"
	      }
	    ],
	  },
	}

Reply:
Yeah, this annoyed me a little too, I too appreciate coherence and all, but then I got lazy... :-).. i'll put some energy into it...